### DISCUSSION TITLE
Average build and rebuild time

### DISCUSSION THREAD

**Discussion by slo-loris** [wailsapp/wails#2030] [Q&A] [answered]
Hi, I have been using wails for past one week. Firstly Kudos for the Great initiative!! (Decided to move from electronJS which is a resource vampire 😅)

System details:

```
System
------
OS:             Windows 10 Home Single Language
Version:        2009 (Build: 19044)
ID:             21H2
Go Version:     go1.19.2
Platform:       windows
Architecture:   amd64

Wails
------
Version:        v2.1.0

Dependency      Package Name    Status          Version
----------      ------------    ------          -------
WebView2        N/A             Installed       107.0.1418.26
npm             N/A             Installed       8.12.2
*upx            N/A             Available
*nsis           N/A             Available

* - Optional Dependency

Diagnosis
---------
Your system is ready for Wails development!
Optional package(s) installation details:
  - upx : Available at https://upx.github.io/
  - nsis : Available at https://nsis.sourceforge.io/Download
```

Currently the average time taken for initial build while executing wails dev in my PC is about ~2 mins and time taken when rebuild is triggered is only around 10 seconds. I noticed that it the frontend part which takes more time. Am I missing any tag to make the compile time little better or the time I mentioned is standard?

Also I noticed that rebuild is getting triggered even when any go file is saved without any change to the file. Can this behavior be changed to rebuild only when files are actually changing?

**Comment by leaanthony** [comment, marked as answer]
Welcome 👋 compile times vary between machines. A full project build on my machine is usually under 10s. What kind of project do you have? The default templates use vite which don't require any rebuild at all for the frontend. Perhaps open a ticket with all the details you have and offer a suggestion on how it can be improved. We are always looking to improve 👍

**Reply by slo-loris** [reply]
Regarding build time, I agree that it could be because of my machine. My CPU is little rusty one 😅

Regarding the latter part (rebuild trigger) am raising it as a feature request.
